custom-service-adapter/src/main/java/io/knotx/tutorials/CustomServiceAdapter.java

```diff
@@ -0,0 +1,47 @@
+package io.knotx.tutorials;
+
+import io.knotx.proxy.AdapterProxy;
+import io.knotx.tutorials.impl.CustomServiceAdapterProxyImpl;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.MessageConsumer;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import io.vertx.rxjava.ext.jdbc.JDBCClient;
+import io.vertx.serviceproxy.ProxyHelper;
+
+public class CustomServiceAdapter extends AbstractVerticle {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CustomServiceAdapter.class);
+
+  private MessageConsumer<JsonObject> consumer;
+  private CustomAdapterConfiguration configuration;
+
+  @Override
+  public void init(Vertx vertx, Context context) {
+    super.init(vertx, context);
+    configuration = new CustomAdapterConfiguration(config());
+  }
+
+  @Override
+  public void start() throws Exception {
+    LOGGER.info("Starting <{}>", this.getClass().getSimpleName());
+
+    final io.vertx.rxjava.core.Vertx rxVertx = new io.vertx.rxjava.core.Vertx(this.vertx);
```

Comment:
If you extend AbstractVerticle  from rx pacakge when you don't need to wrap vertx object but just use it